### DISCUSSION TITLE
Reset Password in Headless

### DISCUSSION THREAD

**Discussion by ricocsharp** [OrchardCMS/OrchardCore#13034] [Q&A]
Working with a Headless solution is there any endpoint to recover password or something like that ?

Bests

**Comment by hishamco** [comment]
I don't think so, @Skrypt correct me if I'm wrong

**Reply by ricocsharp** [reply]
So how this can be achieved with an extension?

**Reply by hishamco** [reply]
Seems we miss a lot of functionalities in terms of REST APIs

**Comment by ns8482e** [comment]
No there is no endpoint to recover password for admin.  However it's achievable same way as in ASP.NET Identity, as OC implements ASP.NET Identity

**Reply by ricocsharp** [reply]
I'm not talking about admin password, a mean any user password since my Solution is based on Headless CMS and a Angular 13 client app

**Comment by ns8482e** [comment]
It seems that you are looking for self service password reset functionality.

There is no built in feature. However Admin can reset password for user.

You can implement self service password reset functionality using custom module.

You need to implement generating password reset token and/or send token via email.

You can inject password manager in your controller and use `GeneratePasswordResetTokenAsync` to generate token.

```cs

var token = await _userManager.GeneratePasswordResetTokenAsync(user);

```

And send token via email and implement action/url that reads token and reset user's password by calling `ResetPasswordAsync`

```cs

  var resetPassResult = await _userManager. ResetPasswordAsync(user, resetPasswordModel.Token, resetPasswordModel.Password);

```
